File: src/FSharpVSPowerTools.Pure/Commands/LintQuickInfoProvider.fs
module FSharpVSPowerTools.LintQuickInfoProvider

open FSharpVSPowerTools.Linting
open Microsoft.VisualStudio.Language.Intellisense
open Microsoft.VisualStudio.Shell
open Microsoft.VisualStudio.Text
open Microsoft.VisualStudio.Text.Tagging
open Microsoft.VisualStudio.Utilities
open System
open System.ComponentModel.Composition


[<Export (typeof<IQuickInfoSourceProvider>)>]
[<Name ("F# Lint Quick Info Provider")>]
[<Order (Before = "Default Quick Info Presenter")>]
[<ContentType "F#">]

type LintQuickInfoProvider [<ImportingConstructor>]
    //( [<Import(typeof<SVsServiceProvider>)>] 
     //   serviceProvider : IServiceProvider,
    (    viewTagAggregatorFactoryService : IViewTagAggregatorFactoryService ) =

    interface IQuickInfoSourceProvider with
        member __.TryCreateQuickInfoSource textBuffer =
            maybe {
                let! generalOptions = Setting.tryGetGeneralOptions()// serviceProvider
                if not generalOptions.LinterEnabled then return! None else
                return
                    new LintQuickInfoSource (textBuffer, viewTagAggregatorFactoryService)
                        :> IQuickInfoSource
            } |> Option.getOrElse null

File: src/FSharpVSPowerTools/Commands/LintQuickInfoProvider.cs
using FSharpVSPowerTools.Linting;
using Microsoft.VisualStudio.Language.Intellisense;
using Microsoft.VisualStudio.Shell;
using Microsoft.VisualStudio.Text;
using Microsoft.VisualStudio.Text.Editor;
using Microsoft.VisualStudio.Text.Tagging;
using Microsoft.VisualStudio.Utilities;
using System;
using System.ComponentModel.Composition;

namespace FSharpVSPowerTools
{
    [Export(typeof(IQuickInfoSourceProvider))]
    [Name("F# Lint Quick Info Provider")]
    [Order(Before = "Default Quick Info Presenter")]
    [ContentType("F#")]
    [TextViewRole(PredefinedTextViewRoles.Editable)]
    [TextViewRole(PredefinedTextViewRoles.PrimaryDocument)]
    internal class LintQuickInfoProvider : IQuickInfoSourceProvider
    {
        private readonly IServiceProvider _serviceProvider;
        private readonly IViewTagAggregatorFactoryService _viewTagAggregatorFactoryService;
        private readonly IGeneralOptions _generalOptions;


        [ImportingConstructor]
        public LintQuickInfoProvider(
            [Import(typeof(SVsServiceProvider))] IServiceProvider serviceProvider,
            IGeneralOptions generalOptions,
            IViewTagAggregatorFactoryService viewTagAggregatorFactoryService)
        {
            _serviceProvider = serviceProvider;
            _viewTagAggregatorFactoryService = viewTagAggregatorFactoryService;
            _generalOptions = generalOptions;
    }

        public IQuickInfoSource TryCreateQuickInfoSource(ITextBuffer textBuffer)
        {
            //var generalOptions = Setting.getGeneralOptions(_serviceProvider);
            if (_generalOptions == null || !_generalOptions.LinterEnabled) return null;

            return new LintQuickInfoSource(textBuffer, _viewTagAggregatorFactoryService);
        }
    }
}